pkg/scheduler/objects/allocation.go

```diff
@@ -135,9 +151,19 @@ func NewAllocationFromSI(alloc *si.Allocation) *Allocation {
 		allocated:         allocated,
 		nodeID:            nodeID,
 		bindTime:          bindTime,
+		foreign:           foreign,
+		preemptable:       preemptable,
 	}
 }
 
+// NewAllocationFromSIAllocated creates an Allocation where the "allocated" flag is always true,
+// regardless whehether the NodeID if empty or not. Used for testing.
+func NewAllocationFromSIAllocated(siAlloc *si.Allocation) *Allocation {
```

Comment:
I understand the temptation to add this, but I'd rather we didn't.

Reply:
Found a way to work it around